internal/core/command_interceptor.go

```diff
@@ -0,0 +1,32 @@
+package core
+
+import (
+	"context"
+)
+
+func CombineCommandInterceptor(interceptors ...CommandInterceptor) CommandInterceptor {
+	var combinedInterceptors CommandInterceptor
+	for _, interceptor := range interceptors {
+		if interceptor == nil {
+			continue
+		}
+		if combinedInterceptors == nil {
+			combinedInterceptors = interceptor
+			continue
+		}
+
+		previousInterceptor := combinedInterceptors
+		combinedInterceptors = func(ctx context.Context, args interface{}, runner CommandRunner) (interface{}, error) {
+			return previousInterceptor(ctx, args, func(ctx context.Context, arg interface{}) (interface{}, error) {
+				return interceptor(ctx, args, runner)
+			})
+		}
+	}
+	return combinedInterceptors
```

Comment:
This won't work if only one interceptor is added

Reply:
Yes it will there is a test to demonstrate it.

Reply:
This is handled by 
```
if combinedInterceptors == nil {
			combinedInterceptors = interceptor
			continue
		}
```

Reply:
Gotcha